### PR TITLE
Adds assassin holoparasites

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscadd: Adds assassin holoparasites, which do low damage and take full damage, but can go invisible, causing their next attack to do massive damage and ignore armor.
rscadd: Assassin holoparasite stealth will be broken by attacking or taking damage, which will briefly prevent the parasite from recalling.
rscadd: Traitor holoparasite injectors include this parasite type.
/:cl:

Guardians no longer apply passive damaging effects to other guardians that share a summoner with them(they'll still damage guardians that don't, of course)

### CODE REVIEW

**Comment by KazeEspada** [comment]
Rogues do it from behind

**Comment by PKPenguin321** [comment]
Sounds like a straight up better version of ye olde standard parasite

**Comment by ChangelingRain** [comment]
It's better if there's no retaliation but if there is retaliation or someone smacks it before it can attack it's worse, due to taking full damage. Also it can't break walls.

**Comment by Zilenan91** [comment]
Yeah this seems like it's just a better standard parasite

**Comment by Robustin** [comment]
Yup, mostly just a standard parasite upgrade.

Almost nobody used the wall punching, aren't very many places where that's actually useful and all it does is leave a nice bread crumb trail for validhunters to chase you down. Sending out a giant "IM VALID PLEASE MURDER ME" sign to break a wall was rarely a better decision than just carrying a welder+wrench with you.

The armor thing is more important but if someone is firing lasers at you then the smart HP's just recall anyway because (unless youre the ranged one lol) holoparasites rarely have anything to gain from chasing around laser users. I'd rather have an invis HP who can slam dunk a laser nerd before he can react than a big stupid one that has to appear on my person before it can move to attack.

I think there are already too many HP's, but if you wanted to actually make this one feel unique I'd up the damage bonus to 60 and reduce the base damage to 10. That way its better at taking out solo targets, but isn't as good for just running down the hall trying to murder everyone. The 60/10 means the TTK is still the same as standards, but with the benefit of immediately slowing/bleeding while also not having the massive weakness of having to appear on my person and move to my target, giving them ample time to run away. That would let them be the initiator in combat instead of just being the sentient edagger that only pops out after you stun someone to help you murder them.

**Comment by Zilenan91** [comment]
I say just close this personally. We don't need more boring varieties of murdermachine that traitors can order.

**Comment by ChangelingRain** [comment]
>robustin has a good idea

Tbh I think it's kind of less that this is absolutely better than standard but that standard is a pile of crap compared to all the other parasites.

**Comment by Zilenan91** [comment]
Standard is great actually. It's one of the better ones if you want to do nothing but munchkin and kill people. It lets you have a ton of flexibility by being able to break walls, take 50% less damage, have no slowdown, drag you, and do 20 damage every hit. It's just overshadowed by other holos because they're easier to use than the Standard one is.

All this does is remove the less important wallbreak and damage resistance and make it far better and far more boring of a punchghost.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -232,6 +238,21 @@ var/global/list/parasites = list() //all currently existing/living guardians
 		src << "<span class='notice'>You deactivate your light.</span>"
 		SetLuminosity(0)
 
+////////parasite tracking/finding procs
+
+/mob/living/proc/hasparasites() //returns a list of guardians the mob is a summoner for
+	var/list/guardians = list()
```
Comment (duncathan) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: `. = list()`

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 2]
```diff
@@ -232,6 +238,21 @@ var/global/list/parasites = list() //all currently existing/living guardians
...
+	for(var/P in parasites)
+		var/mob/living/simple_animal/hostile/guardian/G = P
+		if(G.summoner == src)
+			guardians |= G
+	return guardians
```
Comment (duncathan) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: `. |= G`
implicit return

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 3]
```diff
@@ -232,6 +238,21 @@ var/global/list/parasites = list() //all currently existing/living guardians
...
+
+/mob/living/simple_animal/hostile/guardian/proc/hasmatchingsummoner(mob/living/simple_animal/hostile/guardian/G) //returns 1 if the summoner matches the target's summoner
+	if(istype(G) && G.summoner && G.summoner == summoner)
+		return 1
+	else
+		return 0
```
Comment (duncathan) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: This whole proc can just be
`return (istype(G) && G.summoner == summoner)`

**Pull request merged**